core/primitives/src/block.rs

```diff
@@ -316,6 +316,12 @@ impl Block {
         let num_approvals: u128 =
             approval_mask.iter().map(|x| if *x { 1u128 } else { 0u128 }).sum();
         let total_weight = prev.inner.total_weight.next(num_approvals);
+        let now = Utc::now();
+        let time = if to_timestamp(now) <= prev.inner.timestamp {
```

Comment:
should we just convert to timstamp here?
e.g. let now = to_timestamp(Utc::now()) / compare or +1 is easier. And pass that to block production.